charts/opentelemetry-collector/templates/config/_agent-config-override.tpl

```diff
@@ -11,13 +11,17 @@ exporters:
 
 service:
   pipelines:
-    {{- if and .Values.telemetry.metrics.enabled .Values.standaloneCollector.enabled }}
+    {{- if .Values.standaloneCollector.enabled }}
+
+    {{- if "opentelemetry-collector.metrics.enabled" }}
     metrics:
       exporters: [otlp]
     {{- end }}
 
-    {{- if and .Values.telemetry.traces.enabled .Values.standaloneCollector.enabled }}
+    {{- if "opentelemetry-collector.traces.enabled" }}
```

Comment:
I think it should be possible to avoid `traces/metrics.enabled` option completely. If user wants to disable traces or metrics pipeline, it should be possible to set `configOverride.service.pipelines.metrics: null` and that should disable the pipeline after merging into default config.

Reply:
This is indeed an alias for if it is defined in root,agent or standalone confit override, does that make sense?

Reply:
I was thinking that we could avoid having this alias and conditions like `if "opentelemetry-collector.traces.enabled"` altogether.

According to the docs https://helm.sh/docs/chart_template_guide/values_files/#deleting-a-default-key, if a dict key is set to `null` in user's values.yaml, then that key should be removed as a result. That would work perfectly for disabling a default pipeline by setting `configOverride.service.pipelines.metrics/traces: null` even if some definitions of unused metrics/traces related components will stay in exporters/receivers section (they will not be initialized anyway). I haven't test it yet tho. 

I can try that as the next step after this PR is merged. And for now we can keep the existing approach.

Reply:
OK thanks for explaining. I've adjusted accordingly, PTAL, if it's good I will update existing README examples.